frontend/restInterfaces/AccountRestInterface.ts

```diff
@@ -0,0 +1,79 @@
+import { BaseRestInterface } from './BaseRestInterface';
+import { IUser } from '../../model/user/IUser';
+
+/**
+ * Implements communication with the Server concerning all Account-requests
+ */
+export class AccountRestInterface extends BaseRestInterface {
+  /**
+   * Sends Login-Data to backend and validates them
+   * @param usernameOrEmail the username or email used to attempt to log in
+   * @param password the password used to attempt to log in
+   * @returns true if login was successfull
+   */
+  public static async verifyLogin(
+    usernameOrEmail: string,
+    password: string
+  ): Promise<boolean> {
+    return await BaseRestInterface.post<boolean>('/account/login', {
+      usernameOrEmail,
+      password,
+    });
+  }
+
+  /**
+   * Sends data for a new Account to backend
+   * @param account the account
+   * @returns the id of the newly created user
+   */
+  public static async addAccount(account: Partial<IUser>): Promise<number> {
+    return await BaseRestInterface.put<number>('/account/', account);
```

Comment:
We dont directly return the number, but rather an object containing the number, so the parameter for the generic should be `<{id: number}>` and add `.id` to the end of the line